src/include/sof/math/exp_fcn_hifi4.h

```diff
@@ -0,0 +1,31 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+#ifndef __SOF_MATH_EXPONENTIAL_HIFI4_H__
+#define __SOF_MATH_EXPONENTIAL_HIFI4_H__
+
+#include <stdint.h>
+#include <stddef.h>
+
+#if defined(__XCC__)
+/* HiFi */
+#include <xtensa/config/core-isa.h>
+#if XCHAL_HAVE_HIFI4 == 1
+#define EXPONENTIAL_HIFI4 1
+
+#else
+#error "No XCHAL_HAVE_HIFIx found. Cannot build XCC EXP - HiFi module."
```

Comment:
Does this cause problem with HiFi3 build that has XCC but no HIFI4?

Reply:
yet another reason not to partially duplicate [exp_fcn_hifi.h](https://github.com/thesofproject/sof/pull/7033/files#diff-e3d5145d9d47f335c1fa1b9eaccd78946878c3fcb340053c82e0279574f51b23) but to include it instead.

Reply:
Why not just include `exp_fcn_hifi.h` here?

Reply:
Sorry, I missed your comment here. The commit only applies to HiFi4 code.